### PR TITLE
Adds sort flag

### LINKED ISSUE #415: Add --sort flag
Now that we have all the sort options, it would be good to have `--sort` option as well.

From gnu ls:

```
      --sort=WORD            sort by WORD instead of name: none (-U), size (-S),
                             time (-t), version (-v), extension (-X)
```

### PR DESCRIPTION
**Pull request by tumdum**
Fixes #415

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -177,8 +180,22 @@ pub fn build() -> App<'static, 'static> {
...
+            Arg::with_name("sort")
+                .long("sort")
+                .multiple(true)
+                .possible_values(&["size", "time", "version", "extension"])
+                .takes_value(true)
+                .value_name("ORDER")
+                .overrides_with("timesort")
+                .overrides_with("sizesort")
+                .overrides_with("extensionsort")
+                .overrides_with("sort")
```
Comment (meain) [inline src/app.rs:196]:
```suggestion
                .overrides_with("versionsort")
```

File: src/app.rs [hunk 2]
```diff
@@ -177,8 +180,22 @@ pub fn build() -> App<'static, 'static> {
...
                 .help("Natural sort of (version) numbers within text"),
         )
+        .arg(
+            Arg::with_name("sort")
+                .long("sort")
+                .multiple(true)
+                .possible_values(&["size", "time", "version", "extension"])
+                .takes_value(true)
+                .value_name("ORDER")
```
Comment (meain) [inline src/app.rs:192]: I think `FIELD` would be a better name here. Or we could use `WORD` which is what gnu ls uses.

**Review by meain — CHANGES_REQUESTED** [review]
Hi, thanks for the PR.
Just had few minor changes.

File: src/app.rs (hunk 2 above)
Comment (tumdum) [reply src/app.rs:192]: Whichever you prefer :)

Comment (meain) [reply src/app.rs:192]: I think we will go with `WORD`. Now that I think about it, `version` does not really fit into `FIELD`.

**Comment by tumdum** [comment]
Looks to me like a random failure of ci but I have no way to retrigger.

**Comment by meain** [comment]
I have triggered a rerun.

**Pull request merged**